README.md

```diff
@@ -121,3 +121,62 @@ For more information on using ember-cli, visit
 
 For more information on using cargo, visit
 [doc.crates.io](http://doc.crates.io/).
+
+## Deploy a read-only, download-API-only mirror
+
+This mirror will function as a read-only duplicate of crates.io's API. You will
+be able to download crates using your index and your mirror, but the crate files
+will still come from crates.io's S3 storage.
+
+Your mirror will not:
+
+- Allow users to sign up/sign in
+- Allow crate publish
+- Keep track of any statistics
+- Display available crates in its UI
+
+### API server
+
+To deploy the API on Heroku, use this button:
+
+[![Deploy](https://www.herokucdn.com/deploy/button.svg)][deploy]
+
+[deploy]: https://heroku.com/deploy
+
+You do not need to fill in any of the optional fields.
+
+### Index mirror
+
+You also need a mirror of the crates.io git index, and your index needs to point
+to your API server.
+
+1. `git clone https://github.com/rust-lang/crates.io-index.git`
+2. Edit the config.json file to point to your API server so it looks like:
+
+    ```json
+    {
+      "dl": "https://[your heroku app name].herokuapp.com/api/v1/crates",
+      "api": "https://[your heroku app name].herokuapp.com/"
+    }
+    ```
+
+3. Commit and push to wherever you will be hosting your index (ex: github,
+    gitlab, an internal git server)
+4. In your cargo project, replace the crates.io source with your source by
```

Comment:
Crap this is still in nightly cargo only, isn't it... I'll look up the versions and specify them.